### DISCUSSION TITLE
Integration of administrative tasks

### DISCUSSION THREAD

**Discussion by martijndeb** [pachli/pachli-android#263] [Ideas]
Currently when moderating accounts to fight spam on an open server, I have the following workflow;
- In notifications see a new suspicious looking signup
- View account that has signed up
- Open web interface (opens profile)
- Open moderation interface for this user
- Choose action
- Confirm action

This workflow could be enhanced if it was possible to perform the following quick actions on a profile inside of the app.

The mastodon interface exposes the following buttons in the interface, which bring you to a details screen to attach a note. We usually leave the note empty and mostly perform the suspend action. Other actions are more useful when handling reported posts/accounts. (That could be a next step).

![image](https://github.com/pachli/pachli-android/assets/60222/e3668005-6546-47d9-b754-c64b4b192871)

![image](https://github.com/pachli/pachli-android/assets/60222/84087a28-c088-483c-969c-83bed39d01f2)

This request can be seen in two steps;
- The addition of quick actions for administrative tasks, mainly suspending spam accounts, and unsuspending them if a mistake was made).
- More advanced implementation of other administrative tasks (e.g. view outstanding reports, trigger custom warning, freezing accounts, viewing admin audit log).

The relative admin api endpoints;
- [Viewing accounts and taking actions on them](https://docs.joinmastodon.org/methods/admin/accounts/)
- [Viewing reports](https://docs.joinmastodon.org/methods/admin/reports/)

**Comment by nikclayton** [comment]
Thanks for that, and the screenshots, that's very helpful.

When you're using the app are you often in "Moderator" mode -- by which I mean it would be helpful if the moderation controls were always visible?

Or would it make more sense if there was a toggle somewhere in the UI (maybe in the left nav menu, or the "..." menu at the top right) that enables moderation mode?

I ask because that's quite a lot of additional UI to add to a relatively small screen, and if you don't need it all the time it might be better to hide it until you explicitly want it.

The API and screenshot above mentions warning presets. Do you use those? If so, what workflow do you use to manage them?

**Reply by martijndeb** [reply]
I can't ofcourse speak for other users, but when visiting profiles that's usually because of new signups. Those require moderation, and that happens about 10-15 times a day currently (it has been more, it has also been less).

If a toggle to show moderation controls can be opened with one tap, that would be good. Having it behind multiple taps decreases the ux imho, but I'll take any improvement :). 

It could also work something like this, which I think correspondents with what you described too:

- Visit profile
- Press "Moderate"
- Controls show

Or:

- Visit profile
- Press "Moderate"
- A hover/popup/picker opens to choose the moderation action

We don't really use the warning notes that much, and as such we don't have the prefilled warnings set up.
